pkg/ingester/ingester_v2.go

```diff
@@ -102,16 +110,25 @@ func (u *userTSDB) StartTime() (int64, error) {
 	return u.db.StartTime()
 }
 
+func (u *userTSDB) setState(from, to tsdbState) bool {
```

Comment:
**suggestion:** Consider changing this function call to be `casState` since this looks to be a compare and swap operation

```suggestion
func (u *userTSDB) casState(from, to tsdbState) bool {
```